### ISSUE TITLE
Proposal: Make unary minus for unsigned types a compile error

### ISSUE THREAD

**Issue by Snektron** [ziglang/zig#5586]
```zig
fn negate(x: u32) u32 {
    return -x;
}
```
This is currently safety-checked undefined behaviour, however, this misstake can already be catched
at compile time.

**Comment by Snektron** [comment]
Another option is to make `-unsigned` return a signed type of higher bitness, so
```zig
fn negate(x: u32) i33 {
  return -x;
}```
would be valid.

**Comment by pixelherodev** [comment]
Compiler error makes more sense IMO. If you want to change signed <-> unsigned, you should have to use an explicit cast.

**Comment by andrewrk** [comment]
Related:  #1770

**Comment by Snektron** [comment]
Closed by #5654
